File: public/docs/_examples/forms/dart/lib/hero_form_component.dart
// #docplaster
// #docregion
// #docregion no-todo
import 'package:angular2/core.dart';

import 'hero.dart';

const List<String> _powers = const [
  'Really Smart',
  'Super Flexible',
  'Super Hot',
  'Weather Changer'
];

@Component(
    selector: 'hero-form',
    templateUrl: 'hero_form_component.html')
class HeroFormComponent {
  List<String> get powers => _powers;
// #docregion submitted
  bool submitted = false;
// #enddocregion submitted
  Hero model = new Hero(18, 'Dr IQ', _powers[0], 'Chuck Overstreet');
// #enddocregion no-todo
  // TODO: Remove this when we're done
  String get diagnostic => 'DIAGNOSTIC: $model';
// #docregion no-todo

// #docregion submitted
  onSubmit() {
    submitted = true;
  }
// #enddocregion submitted
}

File: public/docs/_examples/forms/ts/src/app/hero-form.component.ts
// #docplaster
// #docregion , v1, final
import { Component } from '@angular/core';

import { Hero }    from './hero';

@Component({
  moduleId: module.id,
  selector: 'hero-form',
  templateUrl: './hero-form.component.html'
})
export class HeroFormComponent {

  powers = ['Really Smart', 'Super Flexible',
            'Super Hot', 'Weather Changer'];

  model = new Hero(18, 'Dr IQ', this.powers[0], 'Chuck Overstreet');

  // #docregion submitted
  submitted = false;

  onSubmit() { this.submitted = true; }
  // #enddocregion submitted

  // #enddocregion final
  // TODO: Remove this when we're done
  get diagnostic() { return JSON.stringify(this.model); }
  // #enddocregion v1

  // #docregion final, new-hero
  newHero() {
    this.model = new Hero(42, '', '');
  }
  // #enddocregion final, new-hero

  skyDog(): Hero {
    // #docregion SkyDog
    let myHero =  new Hero(42, 'SkyDog',
                           'Fetch any object at any distance',
                           'Leslie Rollover');
    console.log('My hero is called ' + myHero.name); // "My hero is called SkyDog"
    // #enddocregion SkyDog
    return myHero;
  }

  //////// NOT SHOWN IN DOCS ////////

  // Reveal in html:
  //   Name via form.controls = {{showFormControls(heroForm)}}
  showFormControls(form: any) {
    return form && form.controls['name'] &&
    // #docregion form-controls
    form.controls['name'].value; // Dr. IQ
    // #enddocregion form-controls
  }

  /////////////////////////////

  // #docregion v1, final
}
